src/services/carrier_account_service.js

```diff
@@ -79,12 +87,57 @@ export default (easypostClient) =>
      * @returns {string} - The endpoint to be used for the carrier account creation request.
      */
     static _selectCarrierAccountCreationEndpoint(carrierAccountType) {
-      if (Constants.CARRIER_ACCOUNTS_WITH_CUSTOM_WORKFLOWS.includes(carrierAccountType)) {
+      if (Constants.CARRIER_ACCOUNTS_WITH_CUSTOM_CREATE_WORKFLOWS.includes(carrierAccountType)) {
         return 'carrier_accounts/register';
       }
+      if (Constants.UPS_OAUTH_CARRIER_TYPES.includes(carrierAccountType)) {
+        return 'ups_oauth_registrations';
+      }
       return 'carrier_accounts';
     }
 
+    /**
+     * Wraps the carrier account creation parameters in the correct format based on the type.
+     * @private
+     * @param {string} carrierAccountType - The type of carrier account to be created.
+     * @param {Object} params - The parameters for the carrier account to be created.
+     * @returns {Object} - The wrapped carrier account creation parameters.
+     */
+    static _wrapCarrierAccountCreationParams(carrierAccountType, params) {
+      if (Constants.UPS_OAUTH_CARRIER_TYPES.includes(carrierAccountType)) {
+        return { ups_oauth_registrations: params };
+      }
+      return { carrier_account: params };
+    }
+
+    /**
+     * Returns the correct carrier_account endpoint when updating a record based on the type.
+     * @private
+     * @param {string} carrierAccountType - The type of carrier account to be updated.
+     * @param {string} carrierAccountId - The ID of the carrier account to be updated.
+     * @returns {string} - The endpoint to be used for the carrier account update request.
+     */
+    static _selectCarrierAccountUpdateEndpoint(carrierAccountType, carrierAccountId) {
+      if (Constants.UPS_OAUTH_CARRIER_TYPES.includes(carrierAccountType)) {
+        return `ups_oauth_registrations/${carrierAccountId}`;
+      }
+      return `carrier_accounts/${carrierAccountId}`;
+    }
+
+    /**
+     * Wraps the carrier account update parameters in the correct format based on the type.
+     * @private
+     * @param {string} carrierAccountType - The type of carrier account to be updated.
+     * @param {Object} params - The parameters for the carrier account to be updated.
+     * @returns {Object} - The wrapped carrier account update parameters.
+     */
+    static _wrapCarrierAccountUpdateParams(carrierAccountType, params) {
+      if (Constants.UPS_OAUTH_CARRIER_TYPES.includes(carrierAccountType)) {
+        return { ups_oauth_registrations: params };
+      }
+      return { carrier_account: params };
+    }
```

Comment:
This function does the same thing as `_wrapCarrierAccountCreationParams` above, you can combine them into one function and rename it something like `_wrapCarrierAccountParams`

Reply:
Eh, I'd like to keep it explicit.

Reply:
But these two methods have exactly the same logic and do the same thing.

Reply:
I feel this is a premature optimization. We can always break these out in the future if they differ. We should be opting to always keep code in it's most primitive and simple form and only expand and abstract when it makes sense which is not the case here.